Add plan-first working loop to Copilot instructions (#84532)
dotnet/roslyn: .github/copilot-instructions.md

Skills live in `.github/skills/<skill-name>/SKILL.md` and are auto-discovered by their YAML `description`. Useful ones here include `code-review`, `ci-analysis`, `analyzer-codefix`, `merge-into-branch`, `snap`, and `update-agent-docs`.

## Working Loop (plan first)

For any **non-trivial** change, start with a short plan **before** writing the implementing diff — and surface it so it can be reviewed before a large diff appears. "Non-trivial" means anything that is cross-file or cross-area, touches a public API / diagnostic ID / analyzer, changes behavior (not just a typo/comment/formatting fix), or where the approach isn't obvious. When in doubt, write the plan — it's cheap.

Write the plan to `plan.md` in your session folder (see the session context) and keep it updated at milestones. A plan is a working artifact, not a deliverable: keep it lean.

> **Plan template** (drop the fields that genuinely don't apply):

Reviewer: ```suggestion
**Plan template** (drop unneeded fields):
```

```markdown
## Plan: <short title>

- **Scope:** what this change will do (the intended outcome).

> - **Non-goals:** what this change explicitly will NOT do (out of scope now).

Reviewer: ```suggestion
- **Scope:** what this change will do.
- **Non-goals:** what this change explicitly will NOT do.
```

- **Affected areas:** projects/files/layers touched (e.g. `src/Compilers/CSharp`, matching `.instructions.md`).
- **Approach:** the intended implementation, and any alternatives considered/rejected.
- **Acceptance:** observable done-state — the behavior/tests that prove it works.
- **Validation:** exact build + targeted test commands you'll run (see Build & Test).
```

> For non-trivial work, **post the plan and wait for approval before writing the implementing diff** — the plan is meant to be reviewed while it's still cheap to change, not after a large diff already exists.

Reviewer: ```suggestion
**Post the plan and wait for approval before writing the implementing diff** — the plan is meant to be reviewed now, not after a large diff already exists.
```
